components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java

```diff
@@ -320,5 +336,170 @@ private static AnalyticsSchema readTableSchema(String basePath) throws IOExcepti
+     *
+     * @param analyticsFileSystem
+     * @param node
+     * @param baseDir
+     * @throws IOException
+     */
+    private static void restoreFileStructure(AnalyticsFileSystem analyticsFileSystem, File node, File baseDir) throws IOException {
+
+        //get the relative path
+        final String relativePath = node.getAbsolutePath().substring(baseDir.getParent().length());
+
+        if (node.isDirectory()) {
+            analyticsFileSystem.mkdir(relativePath);
+            String[] subNodes = node.list();
+            for (String filename : subNodes) {
+                restoreFileStructure(analyticsFileSystem, new File(node, filename), baseDir);
+            }
+        } else if (node.isFile()) {
+            OutputStream out = analyticsFileSystem.createOutput(relativePath);
+            byte[] data = readFile(node);
+            out.write(data, 0, data.length);
+            out.flush();
+            out.close();
+        }
+
+    }
+
+    /**
+     * Read the file into a byte array
+     *
+     * @param file
+     * @return
+     * @throws IOException
+     */
+    public static byte[] readFile(File file) throws IOException {
+
+        byte[] buffer = new byte[(int) file.length()];
+        InputStream inputStream = null;
+        try {
+            inputStream = new FileInputStream(file);
+            if (inputStream.read(buffer) == -1) {
+                throw new IOException("EOF reached while trying to read the whole file");
+            }
+        } finally {
+            try {
+                if (inputStream != null)
+                    inputStream.close();
+            } catch (IOException e) {
+            }
+        }
+
+        return buffer;
+    }
+
+    /**
+     * returning a map of properties read from the config
+     *
+     * @param props
+     * @return
+     */
+    private static Map<String, String> convertToMap(AnalyticsDataServiceConfigProperty[] props) {
+        Map<String, String> result = new HashMap<String, String>();
+        for (AnalyticsDataServiceConfigProperty prop : props) {
+            result.put(prop.getName(), prop.getValue());
+        }
+        return result;
+    }
+
+    private static AnalyticsDataServiceConfiguration loadAnalyticsDataServiceConfig()
+            throws AnalyticsException {
+        try {
+            File confFile =
+                    new File(GenericUtils.getAnalyticsConfDirectory() + File.separator +
+                            AnalyticsDataSourceConstants.ANALYTICS_CONF_DIR +
+                            File.separator + ANALYTICS_DS_CONFIG_FILE);
+            if (!confFile.exists()) {
+                throw new AnalyticsException(
+                        "Cannot initalize analytics data service, " +
+                                "the analytics data service configuration file cannot be found at: " +
+                                confFile.getPath());
+            }
+            System.out.printf("conf: " + confFile.getAbsolutePath());
```

Comment:
This will be run as a script by the user separately therefore it would be proper to inform the user there itself.